### PR TITLE
refactor: staking records

### LINKED ISSUE #223: bonded data exist in Staking page, while no bond records in Account Page.
image
image

> Displayed: Bonded 0.799 in Staking Page

> While NO bond record in the Account page

My address: 5D9p3ZX1grjYThXUyQzVmKB4Bu8x9u9r4Jc4MeySZ9zBTkUX

### PR DESCRIPTION
**Pull request by JayJay1024**

### CODE REVIEW

File: src/api-derive/staking/types.ts [hunk 1]
```diff
@@ -88,3 +89,32 @@ type DeriveUnlocking = {
   remainingEras: BN;
   value: Balance;
 };
+
+export type TsInMs = u64;
+
```
Comment (sxlwar) [inline src/api-derive/staking/types.ts:94]: Do these types exist in darwinia.js? @cinlk

File: src/components/account/StakingHistory.tsx [hunk 2]
```diff
@@ -1,30 +1,134 @@
...
+  const [ledger, setLedger] = useState<DarwiniaStakingStructsStakingLedger | null>();
+  const [unbondedDataSource, setUnondedDataSource] = useState<UnbondDataSourceState[]>([]);
+  const [unbondingDataSource, setUnondingDataSource] = useState<UnbondDataSourceState[]>([]);
+
+  useEffect(() => {
+    if (!controllerAccount) {
+      setLedger(null);
+      return;
+    }
+
```
Comment (sxlwar) [inline src/components/account/StakingHistory.tsx:43]: Set default value for unsub? 
```js
import {noop} from 'lodash';

let unsub: () => void = noop
```

File: src/components/account/StakingHistory.tsx [hunk 3]
```diff
@@ -1,30 +1,134 @@
...
+      }
+    });
+
+    ledger?.ktonStakingLock.unbondings.forEach((item) => {
+      if (item.until.ltn(bestNumber || 0)) {
+        ktonUnbondeds.push(item);
+      } else {
+        ktonUnbondings.push(item);
+      }
+    });
```
Comment (sxlwar) [inline src/components/account/StakingHistory.tsx:88]: There are some functions in Lodash to simplify this process. e.g: unzip?

**Review by sxlwar — CHANGES_REQUESTED** [review]

**Comment by ZolaxZola** [comment]
image

- [x] Reward was supposed to be CKTON


- [x] Can we Accurate to 3 or 6 decimal places here

**Comment by ZolaxZola** [comment]
image
The Lock record which needs to be released manually is not found on the LOCKs page
image

File: src/api-derive/staking/types.ts (hunk 1 above)
Comment (JayJay1024) [reply src/api-derive/staking/types.ts:94]: The new version of darwinia.js has this type, but apps doesn't use this version yet

File: src/components/account/StakingHistory.tsx (hunk 2 above)
Comment (JayJay1024) [reply src/components/account/StakingHistory.tsx:43]: recolved

File: src/components/account/StakingHistory.tsx (hunk 3 above)
Comment (JayJay1024) [reply src/components/account/StakingHistory.tsx:88]: It doesn't seem to work. The lengths of arrays are not the same. Let's keep it like this for now and then optimize it later

**Comment by JayJay1024** [comment]
> The Lock record which needs to be released manually is not found on the LOCKs page

The new version shall prevail here

File: src/api-derive/staking/types.ts (hunk 1 above)
Comment (sxlwar) [reply src/api-derive/staking/types.ts:94]: Maybe we moved to darwinia.js as soon as possible to reduce maintenance costs

**Review by sxlwar — APPROVED** [review]

**Pull request merged**